dakara_server/library/management/commands/createworks.py

```diff
@@ -0,0 +1,225 @@
+import os
+import sys
+import logging
+import importlib
+
+from django.core.management.base import BaseCommand, CommandError
+
+from library.models import (WorkType, WorkAlternativeTitle, Work)
+
+from .feed_components import default_work_parser
+
+# get logger
+logger = logging.getLogger(__name__)
+
+
+class WorkCreator:
+    """Work creator and updater. Create and update works in the
+    database provided a work file and a parser.
+
+    Args:
+        work_file (str): Absolute path of the file storing the works data.
+        parser (module): Custom python module used to extract data from file.
+        dry_run (bool): Run script in test mode.
+
+    About parser:
+        This module should define a method called `parse_work` which takes
+        a file path as argument and return a dictionnary with the following:
+            works (dict): keys are a query name worktype, values are
+            dictionnary such that:
+                keys are title of a work associated to the worktype, values are
+                the following:
+                    subtitles (list): list of subtitles of a work
+                    alternative_titles (list): list of alternative names of
+                    a work
+    """
+    def __init__(
+            self,
+            work_file,
+            parser):
+        self.work_file = work_file
+        self.parser = parser
+
+    @staticmethod
+    def check_parser_result(dict_work, work_title=None):
+        """Check if a work is correctly structured
+
+        Return
+            True if the work dictionnary is correctly structured
+        """
+        field_names = ('subtitles', 'alternative_titles')
+
+        if not isinstance(dict_work, dict):
+            logger.warning((
+                "Value associated to key work {} "
+                "should be a dictionnary".format(work_title)))
+            return False
+
+        has_correct_struct = True
+        for field in dict_work:
+            if field not in field_names:
+                has_correct_struct = False
+                if work_title:
+                    logger.warning(
+                        "Incorrect field '{}' for '{}'.".format(
+                            field,
+                            work_title))
```

Comment:
I am not sure it is worth to limit the format of incoming data. For me, as long as `dict_work` has the requested fields, it is OK, as the extra keys won't be used in the code.